lib/get-config.js

```diff
@@ -50,12 +67,124 @@ function migrateRulesFromRoot(config, options) {
   }
 }
 
-function processExtends(config) {
+function processPlugins(config, options) {
+
+  var logger = options.console || console;
+
+  var configPlugins = config.plugins;
+  var pluginsHash = {};
+
+  for (var i = 0; i < configPlugins.length; i++) {
+
+    var plugin = configPlugins[i];
+    if (typeof plugin === 'string') {
+      plugin = readPluginFromDisk(options, plugin);
+    } else if (typeof plugin !== 'object') {
+      logger.log(chalk.yellow('Plugin should be string to the plugin file or inline plugin object'));
+    }
+
+    if (plugin && plugin.name) {
+      pluginsHash[plugin.name] = plugin;
+    } else {
+      logger.log(chalk.yellow('Plugin name property is empty'));
+    }
+
+  }
+
+  config.plugins = pluginsHash;
+
+  return config;
+}
+
+function processLoadedRules(config, options) {
+
+  var logger = options.console || console;
+
+  //load all the default rules in `ember-template-lint`
+  var loadedRules = assign({}, defaultRules);
+
+  //load plugin rules
+  for (var name in config.plugins) {
+
+    var pluginRules = config.plugins[name].rules;
+    if (pluginRules) {
+      loadedRules = assign(loadedRules, pluginRules);
+    } else {
+      logger.log(chalk.yellow('Plugin ' + name + ' does not define rules'));
+    }
+
+  }
+
+  config.loadedRules = loadedRules;
+
+  return config;
+
+}
+
+function processExtends(config, options) {
+
+  var logger = options.console || console;
+
+  function _getExtendedRules(extendsName) {
```

Comment:
Can this be extracted to be a stand alone function (i.e. not nested)?  I can't tell if it requires closure scope access to things for it to function...